### PR TITLE
kdb: fix stash

### LINKED ISSUE #1601: kdb stash: does not remove system keys
# Describe what you wanted to do

`kdb stash`

# Describe what you expected

`kdb ls` only lists hard-coded `system/elektra` keys

# Describe what actually happened

`system/globaltestkey` was still there.

# System Information

- Elektra Version: master

### PR DESCRIPTION
**Pull request by markus2330**
closes #1601

## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [ ] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [ ] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [ ] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] meta data is updated (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))

## Review

Reviewers will usually check the following:

- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)
- [ ] Documentation is concise and good to read
- [ ] Examples are well chosen and understandable

## Labels

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the build server is happy** and also you
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by markus2330** [comment]
@sanssecours In https://cirrus-ci.com/task/5940816453828608 there is following output:
```
if [[ "$ENABLE_ASAN" == 'ON' ]]; then
  ninja run_nocheckshell
elif [[ "$BUILD_FULL" == 'ON' ]]; then
/tmp/scripts487f6999eb9d18a44784045d87f3c67c.sh: [[: not found
/tmp/scripts487f6999eb9d18a44784045d87f3c67c.sh: [[: not found
[here all test cases run successfully]
/tmp/scripts487f6999eb9d18a44784045d87f3c67c.sh: kdb: not found
```
Any idea?

File: scripts/restore [hunk 1]
```diff
@@ -30,8 +30,7 @@ fi
 
 PATHS="$(cat "${TMPDIR}/info")"
 
-$KDB umount-all 2> /dev/null
-$KDB rm -r / 2> /dev/null
+$KDB reset -f
```
Comment (sanssecours) [inline scripts/restore]:
```suggestion
"$KDB" reset -f
```

File: scripts/stash [hunk 2]
```diff
@@ -9,9 +9,11 @@ if [ -z "$KDB" ]; then
...
+fail() {
+	echo "$*" 2> /dev/null
+	exit 1
+}
 
-$KDB umount-all 2> /dev/null
-$KDB rm -r system 2> /dev/null
-$KDB rm -r user 2> /dev/null
-$KDB rm -r spec 2> /dev/null
+$KDB backup || fail "Could not backup successfully, will abort"
```
Comment (sanssecours) [inline scripts/stash]:
```suggestion
"$KDB" backup || fail "Could not backup successfully, will abort"
```

File: scripts/stash [hunk 3]
```diff
@@ -9,9 +9,11 @@ if [ -z "$KDB" ]; then
...
+
+$KDB reset -f
```
Comment (sanssecours) [inline scripts/stash]:
```suggestion
"$KDB" reset -f
```

File: scripts/umount-all [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
+fail() {
+	echo "$*" 2> /dev/null
+	exit 1
+}
+
+if [ "$1" != "-f" ]; then
+	fail "This is a very dangerous operation, read the man page first"
+fi
+
+$KDB rm -r system/elektra/mountpoints
```
Comment (sanssecours) [inline scripts/umount-all]:
```suggestion
"$KDB" rm -r system/elektra/mountpoints
```

File: scripts/umount-all [hunk 5]
```diff
@@ -1,11 +1,22 @@
...
+$KDB rm -r system/elektra/globalplugins
```
Comment (sanssecours) [inline scripts/umount-all]:
```suggestion
"$KDB" rm -r system/elektra/globalplugins
```

**Comment by markus2330** [comment]
> I think we can just use [ instead.

Ok, I added 5216814

> The real issue seems to be that the command kdb could not be found

But where is it called? Is it also from the cirrus script?

Btw. is it possible to commit all suggested changes into one commit?

**Comment by sanssecours** [comment]
> But where is it called?

https://github.com/ElektraInitiative/libelektra/blob/c3ccde88419332dd72bfd60969116adda5d4740b/.cirrus.yml#L47-L49

> Is it also from the cirrus script?

Yes.

> Btw. is it possible to commit all suggested changes into one commit?

Yes, you just need to use the button “Add suggestion to batch”, as mentioned [here](https://help.github.com/en/articles/incorporating-feedback-in-your-pull-request).

**Comment by markus2330** [comment]
Actually it had nothing to do with `kdb` not found but the failure was already in the step before (man page was not found and ronn was not installed). Strange that it is so difficult for CI systems to highlight the correct step (our Jenkins also often has the same problem).

**Pull request merged**